### PR TITLE
Support markdown formatting on reports' text property

### LINKED ISSUE #519: Generate Markdown in Reports
When you use the fetch endpoint for parsing news articles, the returned text is a single big block of text.

### PR DESCRIPTION
**Pull request by cesarvarela**
This turned into adding support for markdown on the report's text property.

- `reports.text` and `submissions.text` is now stored as markdown
- `plain_text` field added to `submissions` and `reports` collections
- `text` field is pushed as plain text to the Algolia index
- translations store `text` as markdown since from my tests Google's Translate API can handle it well
- a migration updates every `submission` and `report` to normalizes the newline handling

- added a markdown editor 
image

(the custom resolver option was trickier because realm doesn't handle external npm dependencies very well)

_tests wont's pass until migration runs_

closes #519

### CODE REVIEW

**Comment by smcgregor** [comment]
Blocking

* Merging this: https://github.com/responsible-ai-collaborative/aiid/pull/535 it is fixing a current issue in production so I won't add something that could block deploying from staging.

Not blocking

* We can't use the Google API to translate Markdown. I don't believe it will process links properly. For instance, how will the translation API know which word or words in the translation corresponds to the link? This may be an issue we need to open rather than blocking this from rolling forward in the short term. We can choose to strip links, etc. up until we have a solution.
* It looks like it is reinterpreting the currently plain text database as markdown, which should generally work but there is a possibility of something hiding among the reports that is not intended to be markdown. We will need to check for weird things and weird formatting across the whole text database.
* People are consuming the text of the project for natural language processing (NLP) purposes. Those models should not be processing markdown content. Can the GraphQL endpoint optionally return markdown or plaintext?
* People need to be told in the submission form that they can submit markdown
* At some point in the future we may want to re-parse the historical reports to have its original formatting, this should look significantly better.

Other Comments

* Displaying non-markdown content in the discover app makes sense. People will click through to view with formatting.
* This should merge after addressing the one blocking issue.
* This comment is formatted with Markdown ;)

**Comment by cesarvarela** [comment]
Updated

**Comment by smcgregor** [comment]
@cesarvarela I see this is marked for review. Should I review it? Builds are currently failing.

File: site/gatsby-site/src/components/taxa/Taxonomy.js [hunk 1]
```diff
@@ -57,10 +57,6 @@ const Taxonomy = ({ taxonomy, incidentId }) => {
   const canEdit =
     isRole('taxonomy_editor') || isRole('taxonomy_editor_' + taxonomy.namespace.toLowerCase());
 
-  if (!canEdit && taxonomy.classificationsArray.length < 1) {
```
Comment (smcgregor) [inline site/gatsby-site/src/components/taxa/Taxonomy.js:60]: Why did you remove this? I think this makes the taxonomy not show up in the event no values are associated with the taxonomy. This is the desired behavior.

File: site/gatsby-site/src/global.css [hunk 2]
```diff
@@ -911,3 +911,16 @@ img {
     padding-left: 0px;
   }
 }
+
+/* disable Github Permalink */
+.bytemd-toolbar-right [bytemd-tippy-path='5'] {
```
Comment (smcgregor) [inline site/gatsby-site/src/global.css:916]: What is this doing?

File: site/gatsby-site/src/utils/AlgoliaUpdater.js [hunk 3]
```diff
@@ -59,9 +63,12 @@ class AlgoliaUpdater {
       for (const report_number of incident.reports) {
         const report = reports.find((r) => r.report_number == report_number);
 
+        const text = await remark().use(remarkStrip).process(report.text);
```
Comment (smcgregor) [inline site/gatsby-site/src/utils/AlgoliaUpdater.js:66]: Should use the plain_text field instead, no?

File: site/gatsby-site/src/utils/Translator.js [hunk 4]
```diff
@@ -83,9 +85,15 @@ class Translator {
...
       .collection(`incident_reports_${language}`);
 
-    const fields = [...keys, 'report_number'];
+    const translated = [];
 
-    const translated = items.map((item) => lodash.pick(item, fields));
+    for (const item of items) {
+      const { report_number, text, title } = item;
+
+      const plain_text = (await remark().use(remarkStrip).process(text)).contents.toString();
```
Comment (smcgregor) [inline site/gatsby-site/src/utils/Translator.js:93]: Why not using the plain_text field here? Also, I thought translations used the markdown...

File: site/gatsby-site/src/utils/discoverIndexGenerator.js [hunk 5]
```diff
@@ -146,13 +150,16 @@ module.exports = async ({ graphql }) => {
         }
       }
 
+      const text = await remark().use(remarkStrip).process(report.text);
```
Comment (smcgregor) [inline site/gatsby-site/src/utils/discoverIndexGenerator.js:153]: Same here, why not plain_text?

File: site/gatsby-site/src/components/submissions/SubmissionForm.js [hunk 6]
```diff
@@ -110,17 +113,9 @@ const SubmissionForm = () => {
 
   const [parsingNews, setParsingNews] = useState(false);
 
-  const coldStartToast = () => {
-    addToast({
```
Comment (smcgregor) [inline site/gatsby-site/src/components/submissions/SubmissionForm.js:114]: Cold starts are no longer a problem then?

**Comment by smcgregor** [comment]
A few questions/comments inline, otherwise 👍

File: site/gatsby-site/src/global.css (hunk 2 above)
Comment (cesarvarela) [reply site/gatsby-site/src/global.css:916]: The Markdown Editor has a button that links to the GitHub project, and that's the only way to hide it at the moment.

File: site/gatsby-site/src/components/taxa/Taxonomy.js (hunk 1 above)
Comment (smcgregor) [reply site/gatsby-site/src/components/taxa/Taxonomy.js:60]: We are going to have a few of these taxonomies through time and we don't want to show them if they are blank and the user can't fill them in. I would take a re-render over displaying it to all users.

Comment (cesarvarela) [reply site/gatsby-site/src/components/taxa/Taxonomy.js:60]: Oh got it 👍

File: site/gatsby-site/src/utils/AlgoliaUpdater.js (hunk 3 above)
Comment (cesarvarela) [reply site/gatsby-site/src/utils/AlgoliaUpdater.js:66]: Checked again and no, because that `report` object is built from the `reports` collection + the translated `text`. This will need to be updated later to take into account the `report.language` field though.

File: site/gatsby-site/src/utils/Translator.js (hunk 4 above)
Comment (cesarvarela) [reply site/gatsby-site/src/utils/Translator.js:93]: That is storing the already translated text

File: site/gatsby-site/src/components/submissions/SubmissionForm.js (hunk 6 above)
Comment (cesarvarela) [reply site/gatsby-site/src/components/submissions/SubmissionForm.js:114]: Yup

File: site/gatsby-site/src/utils/discoverIndexGenerator.js (hunk 5 above)
Comment (cesarvarela) [reply site/gatsby-site/src/utils/discoverIndexGenerator.js]: updated

File: site/gatsby-site/src/components/submissions/SubmissionForm.js (hunk 6 above)
Comment (cesarvarela) [reply site/gatsby-site/src/components/submissions/SubmissionForm.js:114]: I mean, it when from the 20 to 30 secs to only a few.

File: site/gatsby-site/src/components/taxa/Taxonomy.js (hunk 1 above)
Comment (cesarvarela) [reply site/gatsby-site/src/components/taxa/Taxonomy.js:60]: Updated, refactored it a little bit, the re-render will only happen to authenticated users.

**Comment by cesarvarela** [comment]
Updated. Tests won't pass until the migration is run.

**Comment by smcgregor** [comment]
@cesarvarela It looks like submit.js and submitted.js conflict now. Can you resolve and we will merge Wednesday?
